tasks/create_config.yml

```diff
@@ -1,5 +1,80 @@
 ---
 
+- name: Set passthrough vars for qdr_config role
+  set_fact:
+    addresses: "{{ qdr_addresses }}"
+    autolink_addresses: "{{ qdr_autolink_addresses }}"
+    connectors: "{{ qdr_connectors }}"
+    extra_listeners: "{{ qdr_extra_listeners }}"
+    listener_auth_peer: "{{ qdr_listener_auth_peer }}"
+    listener_port: "{{ qdr_listener_port }}"
+    listener_require_ssl: "{{ qdr_listener_require_ssl }}"
+    listener_sasl_mech: "{{ qdr_listener_sasl_mech }}"
+    listener_ssl_cert_db: "{{ qdr_listener_ssl_cert_db }}"
+    listener_ssl_cert_file: "{{ qdr_listener_ssl_cert_file }}"
+    listener_ssl_key_file: "{{ qdr_listener_ssl_key_file }}"
+    log_output: "{{ qdr_log_output }}"
+    password: "{{ qdr_password }}"
+    ssl_password: "{{ qdr_listener_ssl_password }}"
+    ssl_profiles: "{{ qdr_ssl_profiles }}"
+    ssl_pw_file: "{{ qdr_listener_ssl_pw_file }}"
+    trusted_certs: "{{ qdr_listener_trusted_certs }}"
+    username: "{{ qdr_username }}"
+
+# *CS
+# Q1. Alex Schultz suggested I consider moving this back to THT. It's so much more readable and testable here.
+#     Comments?
```

Comment:
For other readers, I was asking about L31 in this question, but Emma was answering about L5-22.

L5-22 exists in order to make the vars mutable in this play (to add/remove connectors and ssl profiles, or any other future config that might be done in this layer), as well as to adjust some names for backwards compatibility with OSP16 config. 

The mutability issue is because these are passed in from THT via include_role vars which makes them immutable to adjustment here. 

I will test out some other solutions, but matching the THT var names directly to the qdr-config role names is not really possible without breaking backwards compatibility, they didn't match the puppet ones either.

Reply:
@elfiesmelfie would this be more palatable if I simply moved these mappings to the include_role task below? I don't immediately see a solution to my mutability problem unless I break every piece of config into two parts (a "base" part from THT and an "extra" part generated here) and then implement logic to selectively discard the base part (connectors), or merge them together (ssl profiles) at the include_role task. I'm not sure that will be any more clear, though.

For reference, here is the THT side, which looks almost exactly like your collectd one: https://github.com/csibbitt/tripleo-heat-templates/blob/e9c57ec39f031f1815d3903877faf2bab20f103f/deployment/metrics/qdr-container-ansible.yaml#L174

Reply:
For extra clarity - are you suggesting that your preference is if the variable listed in THT[1]  match the names and set of variables in the config playbook [2] ?

[1] https://github.com/csibbitt/tripleo-heat-templates/blob/e9c57ec39f031f1815d3903877faf2bab20f103f/deployment/metrics/qdr-container-ansible.yaml#L174exactly

[2] https://github.com/infrawatch/qdr-config-ansible-role/blob/master/defaults/main/main.yml

Reply:
I cleaned this up so that it's clearer which vars are being passed through directly  (just stripped of qdr_ prefix) and which are potentially altered by this play. Not sure that achieves what you're asking for, but it should help.